posts/inside-rust/2024-08-12-announcing-metrics-initiative.md

```diff
@@ -0,0 +1,23 @@
+---
+layout: post
+title: "Announcing the Metrics Initiative"
+author: Jane Losare-Lusby and Esteban Kuber
+team: The Rust Compiler Team <https://www.rust-lang.org/governance/teams/compiler>
+---
+
+We ([Jane](https://github.com/yaahc/) and [Esteban](https://github.com/estebank)) are excited to announce the start of the Metrics initiative.
+
+We're envisioning three use cases for metrics within rustc.
+
+* Supporting feature development
+    * e.g., answering specific questions such as when the old and new trait solvers diverge or helping identify and resolve bugs before impacting users
+* Guiding improvements to User Experience,
+    * e.g., knowing which compiler errors are causing the most confusion or are hit the most frequently, focusing on improving those first, and verifying that the improvements help
+* Improving perf feedback loops and insight,
+    * e.g., helping identify pathological edge cases, similar to work @nnethercote has done manually in the past
+
+We're at the point of the initiative where we would like to inform the project members about it and start implementing the metrics infrastructure in collaboration with their real-world needs.
```

Comment:
Discussion: would we want to explicitly mention here that the initiative fully intends to respect user privacy (i.e. no automated uploads / networking)

> NO TELEMETRY, NO NETWORK CONNECTIONS

as mentioned in the tracking issue?

Reply:
My preference is to try to direct people toward the tracking issue and keep the blog post as short and concise as possible.

Reply:
I think it's incredibly important to mention this in the first sentence in the blog post. If we don't say this clearly, people will get the wrong impression. Most people won't care enough to read the tracking issue or anything other than the blog post, and they shouldn't get wrong impressions from it.

Reply:
Die example,
>  to announce the start of the Metrics initiative, a privacy-respecting way to get user feedback.

"metrics" sounds very ambiguous

Reply:
I've applied the suggested wording from @Kobzol below which I believe resolves this concern